tests/e2e.bats

```diff
@@ -3,14 +3,33 @@
 load 'test_helper/bats-support/load'
 load 'test_helper/bats-assert/load'
 
+# ---- GLOBAL CLEANUP ----
 teardown() {
```

Comment:
This is an additional request, but in teardown, can we try to use `BATS_TEST_COMPLETED` to log the YAMLs (kubectl get -o yaml) for all DeviceClass, ResourceSlice and ResourceClaim on failures. And also, log the `kubectl describe` for all pods.